development/Dockerfile

```diff
@@ -1,45 +1,73 @@
+FROM ghcr.io/nautobot/nautobot-dev:${NAUTOBOT_VER}-py${PYTHON_VER}
+
+# Runtime argument and environment setup
 ARG NAUTOBOT_ROOT=/opt/nautobot
 
+ENV prometheus_multiproc_dir=/prom_cache
 ENV NAUTOBOT_ROOT ${NAUTOBOT_ROOT}
 
-RUN apt-get update && apt-get install -y gcc libmariadb-dev-compat libmariadb-dev libgeos-dev
-
-WORKDIR $NAUTOBOT_ROOT
-
-# Configure poetry
+# Don't need virtual environments in container, parallelized installer has inconsistent behavior
 RUN poetry config virtualenvs.create false \
     && poetry config installer.parallel false
 
+# !!! USE CAUTION WHEN MODIFYING LINES ABOVE
 # -------------------------------------------------------------------------------------
-# Install Nautobot Plugin
+# Plugin-specific system build/test dependencies.
+# 
+# Example: LDAP requires `libldap2-dev` to be apt-installed before the Python package.
 # -------------------------------------------------------------------------------------
-WORKDIR /tmp/install
-
-# Copy in only pyproject.toml/poetry.lock to help with caching this layer if no updates to dependencies
-COPY poetry.lock pyproject.toml /tmp/install/
+# --> Start safe to modify section
 
-# Add the requested Nautobot version to pyproject
-# to install the correct version based on the NAUTOBOT_VER argument
-# Otherwise Poetry will override the version in this container
-# with the one in the poetry.lock
-# Redifine NAUTOBOT_VER as a build arg as initial is defined outside FROM.
-ARG NAUTOBOT_VER="1.4.7"
-RUN poetry add nautobot==$NAUTOBOT_VER
+# Uncomment the line below if you are apt-installing any package.
+# RUN apt update
+# RUN apt install libldap2-dev
 
-# --no-root declares not to install the project package since we're wanting to take advantage of caching dependency installation
-# and the project is copied in and installed after this step
-RUN poetry install --no-interaction --no-ansi --no-root -E metrics
+# --> Stop safe to modify section
+# -------------------------------------------------------------------------------------
+# Install Nautobot Plugin
+# -------------------------------------------------------------------------------------
+# !!! USE CAUTION WHEN MODIFYING LINES BELOW
 
-# Copy in the rest of the source code and install local Nautobot plugin
+# Copy in the source code
 WORKDIR /source
 COPY . /source
-RUN cp /tmp/install/* /source/
-RUN poetry install --no-interaction --no-ansi -E metrics
 
-COPY development/nautobot_config.py /opt/nautobot/nautobot_config.py
+# Get container's installed Nautobot version as a forced constraint
+# NAUTOBOT_VER may be a branch name and not a published release therefor we need to get the installed version
+#  so pip can use it to recognize local constraints.
+RUN pip show nautobot | grep "^Version: " | sed -e 's/Version: /nautobot==/' > constraints.txt
+
+# Use Poetry to grab dev dependencies from the lock file
+# Can be improved in Poetry 1.2 which allows `poetry install --only dev`
```

Comment:
why not using poetry 1.2 then?

Reply:
Older Nautobot-dev Docker images come with poetry 1.1.x. IIRC we'd have to explicitly upgrade poetry to latest as part of this Dockerfile, though that might be desirable in any case?

Reply:
Glenn's point is correct. Is this blocking getting started with automated testing?

Reply:
nope, just informational question